Content.Server/Paper/PaperSystem.cs

```diff
@@ -106,22 +109,33 @@ private void OnExamined(EntityUid uid, PaperComponent paperComp, ExaminedEvent a
 
         private void OnInteractUsing(EntityUid uid, PaperComponent paperComp, InteractUsingEvent args)
         {
-            if (_tagSystem.HasTag(args.Used, "Write") && paperComp.StampedBy.Count == 0)
+            // If a pen, attempt to use on paper
+            if (TryComp<PenComponent>(args.Used, out var penComp))
             {
-                var writeEvent = new PaperWriteEvent(uid, args.User);
-                RaiseLocalEvent(args.Used, ref writeEvent);
-                if (!TryComp<ActorComponent>(args.User, out var actor))
-                    return;
-
-                paperComp.Mode = PaperAction.Write;
-                _uiSystem.TryOpen(uid, PaperUiKey.Key, actor.PlayerSession);
-                UpdateUserInterface(uid, paperComp, actor.PlayerSession);
-                return;
+                // If a pen in write mod, write
+                if (penComp.Pen == PenMode.PenWrite)
+                {
+                    if (_tagSystem.HasTag(args.Used, "Write") && paperComp.StampedBy.Count == 0)
```

Comment:
ok so I thought about it, and you should re-arrange the checks. So basically, you should check for your new PenMode first to see if its a stamp, and if it is stamp and return. Otherwise, you can proceed as usual. Basically just intercepting the rest of the actions if it has PenMode.PenStamp or whatever byte it is

Reply:
How about now?